lib/src/zenohRpcClient.cpp

```diff
@@ -112,121 +117,215 @@ UStatus ZenohRpcClient::term() noexcept {
+    std::future<RpcResponse> future;
     z_owned_bytes_map_t map = z_bytes_map_new();
+    z_get_options_t opts = z_get_options_default();
+    z_owned_reply_channel_t *channel = nullptr;
+    UStatus status;
 
-    opts.timeout_ms = requestTimeoutMs_;
-    opts.attachment = z_bytes_map_as_attachment(&map);
+    do {
 
-    if ((0 != payload.size()) && (nullptr != payload.data())) {
-        opts.value.payload.len =  payload.size();
-        opts.value.payload.start = payload.data();
-    } else {
-        opts.value.payload.len = 0;
-        opts.value.payload.start = nullptr;
-    }
+        status.set_code(UCode::INTERNAL);
 
-    z_bytes_map_insert_by_alias(&map, z_bytes_new("header"), bytes);
+        auto uriHash = std::hash<std::string>{}(LongUriSerializer::serialize(topic));
+        auto uuid = Uuidv8Factory::create();
+    
+        UAttributesBuilder builder(uuid, UMessageType::UMESSAGE_TYPE_REQUEST, UPriority::UPRIORITY_CS0);
 
-    if (0 != z_get(z_loan(session_), z_keyexpr(std::to_string(uriHash).c_str()), "", z_move(channel->send), &opts)) {
-        spdlog::error("z_get failure");
-        z_drop(&map);
-        return future;
-    }
+        if (options.has_ttl()) {
+            builder.setTTL(options.ttl());
+            opts.timeout_ms = options.ttl();
+        } else {
+            opts.timeout_ms = requestTimeoutMs_;
+        }
+
+        builder.setPriority(options.priority());
+
+        UAttributes attributes = builder.build();
 
-    future = threadPool_->submit(handleReply, channel);
+        // Serialize UAttributes
+        size_t attrSize = attributes.ByteSizeLong();
+        std::vector<uint8_t> serializedAttributes(attrSize);
+        if (!attributes.SerializeToArray(serializedAttributes.data(), attrSize)) {
+            spdlog::error("attributes SerializeToArray failure");
+            break;
+        }
+
+        z_bytes_t bytes = {.len = serializedAttributes.size(), .start = serializedAttributes.data()};
+        
+        z_bytes_map_insert_by_alias(&map, z_bytes_new("attributes"), bytes);
+
+        channel = new z_owned_reply_channel_t;
+        if (nullptr == channel) {
+            spdlog::error("failed to allocate channel");
+            break;
+        }
+
+        *channel = zc_reply_fifo_new(16);
+
+        opts.attachment = z_bytes_map_as_attachment(&map);
+
+        if ((0 != payload.size()) && (nullptr != payload.data())) {
+            opts.value.payload.len =  payload.size();
+            opts.value.payload.start = payload.data();
+        } else {
+            opts.value.payload.len = 0;
+            opts.value.payload.start = nullptr;
+        }
+        
+        if (0 != z_get(z_loan(session_), z_keyexpr(std::to_string(uriHash).c_str()), "", z_move(channel->send), &opts)) {
+            spdlog::error("z_get failure");
+            break;
+        }
+       
+        future = threadPool_->submit([=] { return handleReply(channel, callback); });
```

Comment:
I think the intent of capturing `[=]` here is to get a copy of `this` `channel` and `callback`. However, implicit capture of `this` is deprecated starting in C++20. Additionally, there are lots of values here that don't seem worth capturing (although I would assume the compiler is smart enough to ignore them). I'd recommend using explicit captures:

`[this, channel, callback] { return handleReply(channel, callback); });`